tests/api/handlers/test_admin.py

```diff
@@ -11,3 +16,104 @@ def test_admin_rules_forbidden(self, client, api_identity):
         api_identity.admin = False
         response = client.get(f"{self.path}/rules")
         assert response.status_code == status.HTTP_403_FORBIDDEN
+
+    def test_list_disabled_rules(self, client, api_identity, db_rule, db_rule_disabled):
+        api_identity.admin = True
+
+        response = client.get(f"{self.path}/rules")
+        assert response.status_code == status.HTTP_200_OK
+        result = response.json()
+        assert len(result) == 1
+        assert result[0] == {
+            "id": 2,
+            "name": "disabledrule",
+            "disabled": True,
+            "user": {"id": 1, "name": "testuser", "is_admin": False},
+            "tracking_rule": {"name": "users-followed", "params": ["user1", "user2"]},
+            "generation_rules": [
+                {
+                    "destinations": [{"protocol": "email", "address": "dude@mcpants"}],
+                    "filters": {
+                        "applications": [],
+                        "severities": [],
+                        "topic": None,
+                        "my_actions": False,
+                    },
+                }
+            ],
+            "generated_last_week": 0,
+        }
+
+    @mock.patch("fmn.api.handlers.admin.publish")
+    def test_reenable_rule(self, publish, client, api_identity, db_rule, db_rule_disabled):
+        api_identity.admin = True
+
+        # First check we have one disabled rule
+        response = client.get(f"{self.path}/rules")
+        assert response.status_code == status.HTTP_200_OK
+        assert len(response.json()) == 1
+
+        # Next re-enable the rule
+        response = client.patch(
+            f"{self.path}/rules/2/disabled", content=json.dumps({"id": 2, "disabled": False})
+        )
+        assert response.status_code == status.HTTP_200_OK
+        result = response.json()
+        assert result["name"] == "disabledrule"
+        assert result["disabled"] is False
+
+        # Finally check we have no disabled rules anymore
+        response = client.get(f"{self.path}/rules")
+        assert response.status_code == status.HTTP_200_OK
+        assert len(response.json()) == 0
+
+        edited_rule = api_models.Rule.from_orm(db_rule_disabled)
+        edited_rule.disabled = False
+        success_message = RuleUpdateV1(
+            body={
+                "rule": edited_rule.dict(),
+                "user": api_models.User.from_orm(db_rule_disabled.user).dict(),
+            }
+        )
+        publish.assert_awaited_once_with(success_message)
+
+    @mock.patch("fmn.api.handlers.admin.publish")
```

Comment:
Same here (about `fedora_messaging.testing.mock_sends()`)

Reply:
oh, i was copying something already in place in handlers/test_users.py

Reply:
that said, still changing it over to fedora_messaging.testing.mock_sends()